platform/viewer/src/components/ViewportGrid/ViewportGrid.js

```diff
@@ -0,0 +1,186 @@
+    // TODO: Probably actually don't need one per element; or to use global state
+    const childrenWithProps = React.Children.map(children, (child, index) => {
+      return React.cloneElement(child, {
+        viewportIndex: viewportIndex,
+        key: index,
+      });
+    });
+    const displaySetStack = _getDisplaySetStackFromStudies(
+      studies,
+      studyInstanceUid,
+      displaySetInstanceUid
+    );
+    const viewportProps = displaySetStack ? { ...displaySetStack } : {};
+    const ViewportComponent = displaySetStack
+      ? getViewportComponent(plugin)
+      : EmptyViewport;
+
+    return (
+      <ViewportPane
+        onDrop={setViewportData}
+        viewportIndex={viewportIndex} // Needed by `setViewportData`
+        className={classNames('viewport-container', {
+          active: activeViewportIndex === viewportIndex,
+        })}
+        key={viewportIndex}
+      >
+        <ViewportComponent
+          // Different "Viewport Types"? Feels too rigid to lock all into these props
+          {...viewportProps}
+          // We shouldn't need this?
+          // Used in `ConnectedCornerstoneViewport`
+          viewportIndex={viewportIndex}
+          setViewportActive={() => {
+            if (activeViewportIndex !== viewportIndex) {
+              setViewportActive(viewportIndex);
+            }
+          }}
+        />
+        {childrenWithProps}
+      </ViewportPane>
+    );
+  });
+
+  const rowSize = 100 / numRows;
+  const colSize = 100 / numColumns;
+
+  // http://grid.malven.co/
+  return (
+    <div
+      style={{
+        display: 'grid',
+        gridTemplateRows: `repeat(${numRows}, ${rowSize}%)`,
+        gridTemplateColumns: `repeat(${numColumns}, ${colSize}%)`,
+        height: '100%',
+        width: '100%',
+      }}
+    >
+      {magic}
+    </div>
+  );
+};
+
+const noop = () => {};
+
+ViewportGrid.propTypes = {
+  activeViewportIndex: PropTypes.number.isRequired,
+  numRows: PropTypes.number.isRequired,
+  numColumns: PropTypes.number.isRequired,
+  viewportPanes: PropTypes.arrayOf(
+    PropTypes.shape({
+      studyInstanceUid: PropTypes.string,
+      displaySetInstanceUid: PropTypes.string,
+      plugin: PropTypes.string,
+    })
+  ).isRequired,
+  availablePlugins: PropTypes.object.isRequired,
+  setViewportData: PropTypes.func,
+  setViewportActive: PropTypes.func, // Connected
+  studies: PropTypes.array,
+  children: PropTypes.node,
+};
+
+ViewportGrid.defaultProps = {
+  activeViewportIndex: 0,
+  availablePlugins: {
+    DefaultViewport,
+  },
+  setViewportActive: noop,
+  defaultPlugin: 'DefaultViewport',
+};
+
+/**
+ *
+ *
+ * @param {*} studies
+ * @param {string} studyInstanceUid
+ * @param {string} displaySetInstanceUid
+ * @returns
+ */
+function _getDisplaySetStackFromStudies(
+  studies,
+  studyInstanceUid,
+  displaySetInstanceUid
+) {
+  if (!studyInstanceUid || !displaySetInstanceUid) {
+    return undefined;
```

Comment:
`return;` is identical, though maybe we should just throw an error

Reply:
There is a fair amount here that we need to talk about. I have a running list of questions around loading data and passing them in an appropriate/consistent way to viewports.